launcher/src/main/java/org/apache/spark/launcher/AbstractCommandBuilder.java

```diff
@@ -182,7 +186,7 @@ void addOptionString(List<String> cmd, String options) {
       addToClassPath(cp, String.format("%s/core/target/jars/*", sparkHome));
     }
 
-    String assembly = findAssembly();
+    final String assembly = AbstractCommandBuilder.class.getProtectionDomain().getCodeSource().getLocation().getPath();
```

Comment:
Can you do just `getClass()` here instead of the more verbose version? Also, line looks pretty long.